discovery-provider/src/utils/redis_metrics.py

```diff
@@ -33,47 +33,48 @@ def get_rounded_date_time():
 def parse_metrics_key(key):
     """
     Validates that a key is correctly formatted and returns
-    the source: (routes|applications) and date of key
+    the source: (routes|applications), ip address, and date of key
     """
     if not metrics_prefix.startswith(metrics_prefix):
         logger.warning(f"Bad redis key inserted w/out metrics prefix {key}")
         return None
 
     fragments = key.split(':')
-    if len(fragments) != 4:
-        logger.warning(f"Bad redis key inserted: must have 4 parts {key}")
+    if len(fragments) != 5:
+        logger.warning(f"Bad redis key inserted: must have 5 parts {key}")
         return None
 
-    _, source, date, time = fragments
+    _, source, ip, date, time = fragments
     if source not in (metrics_routes, metrics_application):
         logger.warning(f"Bad redis key inserted: must be routes or application {key}")
         return None
     date_time = datetime.strptime(f"{date}:{time}", datetime_format)
 
-    return source, date_time
+    return source, ip, date_time
 
 def extract_app_name_key():
     """
     Extracts the application name redis key and hash from the request
     The key should be of format:
-        <metrics_prefix>:<metrics_application>:<rounded_date_time_format>
-        ie: "API_METRICS:applications:2020/08/04:14"
+        <metrics_prefix>:<metrics_application>:<ip>:<rounded_date_time_format>
+        ie: "API_METRICS:applications:192.168.0.1:2020/08/04:14"
     The hash should be of format:
         <app_name>
         ie: "audius_dapp"
     """
     application_name = request.args.get(app_name_param, type=str, default=None)
+    ip = request.remote_addr
     date_time = get_rounded_date_time().strftime(datetime_format)
 
-    appplication_key = f"{metrics_prefix}:{metrics_application}:{date_time}"
-    return (appplication_key, application_name)
+    application_key = f"{metrics_prefix}:{metrics_application}:{ip}:{date_time}"
```

Comment:
so this is by ip by hour? and we delete the key after we process it right? i think this level of redis data should be fine

Reply:
yep!